src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -231,6 +231,7 @@ object Group {
       group is noAppsAndGroupsWithSameName
       group is conditional[Group](_.id.isRoot)(noCyclicDependencies)
       group.groups is every(valid(validNestedGroup(group.id.canonicalPath(base))))
+      group is servicePortsDontOverlap
```

Comment:
Doesn't https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/upgrade/DeploymentPlan.scala#L328 take care of this?